src/wallet/wallet.cpp

```diff
@@ -3960,6 +3967,29 @@ bool CWalletTx::IsImmatureCoinBase() const
     return GetBlocksToMaturity() > 0;
 }
 
+bool CWalletTx::isFinal() const
+{
+    AssertLockHeld(pwallet->cs_wallet);
+    return pwallet->CheckFinalWalletTx(*tx);
+}
+
+bool CWallet::CheckFinalWalletTx(const CTransaction& tx) const
+{
+    AssertLockHeld(cs_wallet);
+    // Wallet not doing historical validation work, it's only
```

Comment:
nit: some grammar improvement on this paragraph

```
    // The wallet is not doing historical validation work, it's only
    // interested in the nLocktime evaluation as enforced on
    // the current chain tip (BIP 113). If there are forks changing
    // this in the future, the wallet should be updated.
```

Reply:
Thanks for review, will take them if I have to update PR!

Reply:
Thanks, took comment.